### DISCUSSION TITLE
size of modulesSpace and ReasonsSpace

### DISCUSSION THREAD

**Discussion by aatifhsn** [webpack/webpack#17336] [Q&A]
I am building a dependency visualiser and analyser tool and I am using statsData in order to get the information about all the modules. I have two queries related to that-
1) For any module, we get the name of the issuer which tells the relationship of a module with other module. However, suppose there are many more relationships of the modules with other modules, one way to check that is the reasons array. Is there any other way through which we can check module relationships?
2) Suppose I am using the reasons array to find the relationships and to build the array but there is a problem that the reasonsSpace and modulesSpace is static(i.e. they have a fixed size). Is there any way through which we can make it dynamic?

**Comment by alexander-akait** [comment]
> For any module, we get the name of the issuer which tells the relationship of a module with other module. However, suppose there are many more relationships of the modules with other modules, one way to check that is the reasons array. Is there any other way through which we can check module relationships?

The module may not have issuer, we have `getIncomingConnections` in `ModuleGraph` to get this, if you write a plugin you can use it

> Suppose I am using the reasons array to find the relationships and to build the array but there is a problem that the reasonsSpace and modulesSpace is static(i.e. they have a fixed size). Is there any way through which we can make it dynamic?

`reasonsSpace: Infinity`, you can also set `stats: "verbose"`

**Reply by aatifhsn** [reply]
If is set the reasonsSpace: Infinity, does it make the array dynamic?

**Reply by alexander-akait** [reply]
Yes, it means put everything and don't look at size

**Reply by aatifhsn** [reply]
But then doesn't it create a problem of size? Like we are allocating a huge amount of space for the stats..

**Reply by alexander-akait** [reply]
It will consome more memory, but it is fine for analize, I don't think you need to run this on the each compilation
